client/client.go

```diff
@@ -1506,6 +1517,9 @@ func (c *Client) watchNodeUpdates(attrHash, metaHash uint64) {
 
 				c.retryRegisterNode()
```

Comment:
IIUC, If this retryRegister method returns without registering because it timed out, it won't try again until the next node update comes through triggerNodeUpdate. Is that desirable?

Reply:
@dadgar and I talked about this. The node should be registered before watching for updates and the functionality that comes below, if I'm understanding this comment correctly.